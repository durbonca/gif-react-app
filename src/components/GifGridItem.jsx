import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {
  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={url} alt={title}/>
      <div className='title-box'>
        <p>{title}</p>
      </div>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}