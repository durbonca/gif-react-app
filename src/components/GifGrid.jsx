import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>

            { loading && <i className='animate__animated flash'>Cargando...</i>}

            <div className='card-grid'>
                {images.map( ( img, i ) =>
                    (
                        <GifGridItem key={i} {...img} />
                        )
                        )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}