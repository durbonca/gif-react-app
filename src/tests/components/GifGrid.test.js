import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('GifGrid Test', () => {

    const category = 'One Punch';
    
    test('snapshot test', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});