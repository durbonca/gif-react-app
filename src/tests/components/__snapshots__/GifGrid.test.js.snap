// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Test snapshot test 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <i
    className="animate__animated flash"
  >
    Cargando...
  </i>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="0"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
